### DISCUSSION TITLE
Count and Sum Repeater

### DISCUSSION THREAD

**Discussion by miftahulrespati** [filamentphp/filament#1329] [Help] [answered]
Is there any feature to count the number of `repeater` and directly access the value of each `repeater` so I can sum the total value? If not, can this be a new feature?
I've attached the image of what I mean. Thanks in advance.
![screen20](https://user-images.githubusercontent.com/63243218/150724501-90bc27e8-0e69-4558-8764-308e02a4a9bd.png)

**Comment by danharrin** [comment, marked as answer]
Yeah, you can try `count($get('repeater_name'))` in any callback - in this case you probably want 2 placeholder fields.

```php
Placeholder::make('total_area')->content(fn ($get) => // calculate the area with $get('repeater_name')),
Placeholder::make('number_of_field')->content(fn ($get) => 'Expected: ' . count($get('repeater_name'))),
```

**Reply by miftahulrespati** [reply]
Thank you! Your answer works.
I've actually been doing this for days. But, instead of `Placeholder`, I use `TextInput->default()` and the `$get` returns a null instead of an array.

```php
TextInput::make('number_of_field')->default(fn ($get) => count($get('repeater_name'))),
```
will throw an error:
```
count(): Argument #1 ($value) must be of type Countable|array, null given 
```

**Reply by diazsasak** [reply]
@miftahulrespati maybe the repeater_name field is not filled? what if check if it null don't count it

**Reply by MPTA** [reply]
i have the same error:
count(): Argument #1 ($value) must be of type Countable|array, null given

**Reply by leandrocfe** [reply]
You should check why your repeater is returning a `null` value
